src/analysis/convex/integral.lean

```diff
@@ -4,51 +4,185 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yury G. Kudryashov
 -/
 import analysis.convex.function
-import measure_theory.integral.set_integral
+import analysis.convex.strict
+import measure_theory.function.ae_eq_of_integral
 
 /-!
 # Jensen's inequality for integrals
 
-In this file we prove four theorems:
+In this file we define `measure_theory.average μ f` (notation: `⨍ x, f x ∂μ`) to be the average
+value of `f` with respect to measure `μ`. It is defined as `(μ univ).to_real⁻¹ • ∫ x, f x ∂μ`, so it
+is equal to zero if `f` is not integrable or if `μ` is an infinite measure. If `μ` is a probability
+measure, then the average of any function is equal to its integral.
 
-* `convex.smul_integral_mem`: if `μ` is a non-zero finite measure on `α`, `s` is a convex closed set
-  in `E`, and `f` is an integrable function sending `μ`-a.e. points to `s`, then the average value
-  of `f` belongs to `s`: `(μ univ).to_real⁻¹ • ∫ x, f x ∂μ ∈ s`. See also `convex.center_mass_mem`
-  for a finite sum version of this lemma.
+Then we prove several forms of Jensen's inequality for integrals.
 
-* `convex.integral_mem`: if `μ` is a probability measure on `α`, `s` is a convex closed set in `E`,
-  and `f` is an integrable function sending `μ`-a.e. points to `s`, then the expected value of `f`
-  belongs to `s`: `∫ x, f x ∂μ ∈ s`. See also `convex.sum_mem` for a finite sum version of this
-  lemma.
+- for convex sets: `convex.average_mem`, `convex.set_average_mem`, `convex.integral_mem`;
 
-* `convex_on.map_smul_integral_le`: Convex Jensen's inequality: If a function `g : E → ℝ` is convex
-  and continuous on a convex closed set `s`, `μ` is a finite non-zero measure on `α`, and
-  `f : α → E` is a function sending `μ`-a.e. points to `s`, then the value of `g` at the average
-  value of `f` is less than or equal to the average value of `g ∘ f` provided that both `f` and
-  `g ∘ f` are integrable. See also `convex_on.map_sum_le` for a finite sum version of this lemma.
+- for convex functions: `convex.on.average_mem_epigraph`, `convex_on.map_average_le`,
+  `convex_on.set_average_mem_epigraph`, `convex_on.map_set_average_le`, `convex_on.map_integral_le`;
 
-* `convex_on.map_integral_le`: Convex Jensen's inequality: If a function `g : E → ℝ` is convex and
-  continuous on a convex closed set `s`, `μ` is a probability measure on `α`, and `f : α → E` is a
-  function sending `μ`-a.e. points to `s`, then the value of `g` at the expected value of `f` is
-  less than or equal to the expected value of `g ∘ f` provided that both `f` and `g ∘ f` are
-  integrable. See also `convex_on.map_sum_le` for a finite sum version of this lemma.
+- for strictly convex sets: `strict_convex.ae_eq_const_or_average_mem_interior`;
+
+- for a closed ball in a strictly convex normed space:
+  `strict_convex.ae_eq_const_or_norm_integral_lt_of_norm_le_const`
+
+- for strictly convex functions: `strict_convex_on.ae_eq_const_or_map_average_lt`.
+
+## TODO
+
+- Add versions for concave functions.
```

Comment:
Do we agree that the generalization isn't immediate because `g : E → ℝ`? What properties are you using from this `ℝ`? [The finite version](https://leanprover-community.github.io/mathlib_docs/analysis/convex/jensen.html#convex_on.map_center_mass_le) only needs `ordered_add_comm_group β` + `ordered_smul ℝ β`

Reply:
I used `concave_on.neg` instead.